### PR TITLE
adds go import statements

### PR DESCRIPTION
**Pull request by rkodev**
1. Adds import statements to go generated snippets.
2. Fixes Me path objects after shim me.

Updated snippet has all necessary imports and resolves wrong imports for shim me path

Resolves https://github.com/microsoftgraph/msgraph-sdk-go/issues/442
Resolves https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/920

```go
//THE GO SDK IS IN PREVIEW. NON-PRODUCTION USE ONLY
import (
	  "context"
	  msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
	  graphconfig "github.com/microsoftgraph/msgraph-sdk-go/users"
	  //other-imports
)

graphClient := msgraphsdk.NewGraphServiceClientWithCredentials(cred, scopes)


requestParameters := &graphconfig.ItemMessagesRequestBuilderGetQueryParameters{
	Select: [] string {"sender","subject"},
}
configuration := &graphconfig.ItemMessagesRequestBuilderGetRequestConfiguration{
	QueryParameters: requestParameters,
}

result, err := graphClient.Me().Messages().Get(context.Background(), configuration)

```

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 1]
```diff
@@ -38,17 +40,106 @@ public string GenerateCodeSnippet(SnippetModel snippetModel)
...
+        {
+            return properties != null && properties.Select(x => searchProperty(x, propertyType)).Where(x => true).FirstOrDefault();
+        }
+
+
+        private static Boolean searchProperty(CodeProperty property, PropertyType propertyType)
+        {
+            if (property.Children != null && property.Children.Any())
+            {
+                var inchildren = property.Children.Select(x => x.PropertyType == propertyType).Where(x => true).FirstOrDefault();
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:134]: same where comment

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 2]
```diff
@@ -163,6 +254,7 @@ private static string GetNestedObjectName(IEnumerable<OpenApiUrlTreeNode> nodes)
                         .Aggregate(static (x, y) =>
                         {
                             var w = x.EndsWith("s") && y.Equals("Item") ? x.Remove(x.Length - 1, 1) : x;
+                            w = w.Equals("Me") ? "Item" : w;
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:257]:
```suggestion
                            w = w.Equals("Me", StringComparison.Ordinal) ? "Item" : w;
```

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for starting this, a couple of comments.

**Review by baywet — DISMISSED** [review]
Thanks for making the changes!

**Review by andrueastman — APPROVED** [review]

**Pull request merged**